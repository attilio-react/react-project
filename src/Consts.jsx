const consts = { 
    SEARCH_BY_TITLE : 0,
    SEARCH_BY_GENRE : 1,

    SORT_BY_RELEASE_DATE : 0,
    SORT_BY_RATING : 1
}

export {consts}
