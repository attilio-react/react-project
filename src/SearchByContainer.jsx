import React from 'react';

import {Label} from './common/Label.jsx';
import {Button} from './common/Button.jsx';

class SearchByContainer extends React.PureComponent {
  render () {
    return <>
		  <Label text='SEARCH BY' />
		  &nbsp;
		  <Button caption='TITLE' selected={true} />
		  &nbsp;
		  <Button caption='GENRE' />
           </>;
  }
}


export {SearchByContainer}


