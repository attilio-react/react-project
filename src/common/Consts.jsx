const consts = { 
    SEARCH_SCREEN : 0,
    DETAIL_SCREEN: 1,


    SEARCH_BY_TITLE : 0,
    SEARCH_BY_GENRE : 1,

    SORT_BY_RELEASE_DATE : 0,
    SORT_BY_RATING : 1,


    PARAM_SORT_BY: 'sortBy',
    PARAM_SORT_ORDER: 'sortOrder',
    PARAM_SEARCH: 'search',
    PARAM_SEARCH_BY: 'searchBy',

    VALUE_BY_GENRE: 'genres',
    VALUE_BY_TITLE: 'title',
    VALUE_BY_RELEASE_DATE: 'release_date',
    VALUE_BY_VOTE_AVERAGE: 'vote_average',

    VALUE_ASC: 'asc',
    VALUE_DESC: 'desc'
}

export {consts}
