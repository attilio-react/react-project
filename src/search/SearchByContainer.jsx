import React from 'react';

import {consts} from 'Common/Consts.jsx'

import {SearchContext} from './SearchContext.jsx'

import {Label} from 'Common/Label.jsx';
import {Button} from 'Common/Button.jsx';

class SearchByContainer extends React.PureComponent {
  render () {
    return (<SearchContext.Consumer>
            {ctx => {
                  const
                    {titleClickCb, genreClickCb, searchBySelection} = ctx,
                    selectTitle = (searchBySelection == consts.SEARCH_BY_TITLE),
                    selectGenre = (searchBySelection == consts.SEARCH_BY_GENRE)
                  
                 return <>
		    <Label text='SEARCH BY' />
		    &nbsp;
		    <Button caption='TITLE' onClick={titleClickCb} selected={selectTitle} />
		    &nbsp;
		    <Button caption='GENRE' onClick={genreClickCb} selected={selectGenre} />
                  </>
            }}
           </SearchContext.Consumer>);
  }
}


export {SearchByContainer}


