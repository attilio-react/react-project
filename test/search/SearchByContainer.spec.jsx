import * as React from 'react';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';

import {SearchByContainer} from '|search/SearchByContainer.jsx';
import {SearchContext} from '|search/SearchContext.jsx';
import {consts} from '|common/Consts.jsx'

const CONTEXT = {
    titleClickCb: jest.fn(),
    genreClickCb: jest.fn()
}

describe('SearchByContainer rendering', () => {
  it('Should render when search by title', () => {
    const 
        component = renderer.create(
            <SearchContext.Provider value={{...CONTEXT, searchBySelection: consts.SEARCH_BY_TITLE}}>
               <SearchByContainer />
            </SearchContext.Provider>),
        json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('Should render when sort by genre', () => {
    const 
        component = renderer.create(
            <SearchContext.Provider value={{...CONTEXT, searchBySelection: consts.SEARCH_BY_GENRE}}>
               <SearchByContainer />
            </SearchContext.Provider>),
        json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

});

describe('SearchByContainer behaviour', () => {
  it('Should call callback when search by title clicked', () => {
    const 
      resultHeader = mount(
            <SearchContext.Provider value={{...CONTEXT, searchBySelection: consts.SEARCH_BY_TITLE}}>
               <SearchByContainer />
            </SearchContext.Provider>),
      nativeButton = resultHeader.find('input').first();
    nativeButton.simulate('click');
    expect(CONTEXT.titleClickCb).toBeCalled()
  });

  it('Should call callback when search by genre clicked', () => {
    const 
      resultHeader = mount(
            <SearchContext.Provider value={{...CONTEXT, searchBySelection: consts.SEARCH_BY_TITLE}}>
               <SearchByContainer />
            </SearchContext.Provider>),
      nativeButton = resultHeader.find('input').at(1);
    nativeButton.simulate('click');
    expect(CONTEXT.genreClickCb).toBeCalled()
  });

});

