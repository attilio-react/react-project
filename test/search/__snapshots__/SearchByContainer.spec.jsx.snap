// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchByContainer rendering Should render when search by title 1`] = `
Array [
  <span>
    SEARCH BY
  </span>,
  " ",
  <input
    onClick={[MockFunction]}
    style={
      Object {
        "textDecoration": "underline",
      }
    }
    type="button"
    value="TITLE"
  />,
  " ",
  <input
    onClick={[MockFunction]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
    type="button"
    value="GENRE"
  />,
]
`;

exports[`SearchByContainer rendering Should render when sort by genre 1`] = `
Array [
  <span>
    SEARCH BY
  </span>,
  " ",
  <input
    onClick={[MockFunction]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
    type="button"
    value="TITLE"
  />,
  " ",
  <input
    onClick={[MockFunction]}
    style={
      Object {
        "textDecoration": "underline",
      }
    }
    type="button"
    value="GENRE"
  />,
]
`;
